core/src/main/scala/org/broadinstitute/dsde/rawls/entities/EntityService.scala

```diff
@@ -515,101 +521,212 @@ class EntityService(protected val ctx: RawlsRequestContext,
-      }
+        // start a transaction; here's where we do a bunch of writes
+        userResult <- dataSource.inTransaction { dataAccess =>
+          val shardId: String = dataAccess.determineShard(workspaceId)
 
-      // get the local (legacy) provider
-      localProvider <- getProviderWithTracing(workspaceContext, ctx)
+          val stopwatch = StopWatch.createStarted()
 
-      // get the list of entity types in this workspace
-      entityTypeMetadata <- localProvider.entityTypeMetadata(useCache = true, ctx)
+          logger.info(
+            s"Quicksilver migration $workspaceId: starting (${stopwatch.formatTime()}) ..."
+          )
 
-      // start a transaction; here's where we do a bunch of writes
-      _ <- dataSource.inTransaction { dataAccess =>
-        val shardId: String = dataAccess.determineShard(workspaceContext.workspaceIdAsUUID)
+          withIncreasedSortMemory(dataAccess) {
+            for {
+              // batch into groups of $batchSize entities, currently 50k. This method returns the min and max entity ids
+              // for each batch. later queries will use those boundaries to migrate entities in batches, which
+              // prevents the temp tables from growing too large.
+              batchBoundaries <- quicksilverCalculateBatches(workspaceId, batchSize, dataAccess)
+              // niceties for logging
+              indexedBoundaries = batchBoundaries.zipWithIndex
+
+              // for each batch, migrate the entities in that batch
+              updateCounts <- DBIO.sequence(indexedBoundaries.map { case (boundary, idx) =>
+                quicksilverMigrateBatch(workspaceId,
+                                        shardId,
+                                        boundary,
+                                        dataAccess,
+                                        idx,
+                                        indexedBoundaries.size,
+                                        stopwatch,
+                                        s
+                )
+              })
+              numEntitiesUpdated = updateCounts.sum
+
+              // populate the ENTITY_REFS table for this workspace
+              _ <- traceDBIOWithParent("migrationAddReferences", s) { _ =>
+                dataAccess.compactEntityQuery.migrationAddReferences(workspaceId, shardId)
+              }
+              _ = logger.info(
+                s"Quicksilver migration $workspaceId: populated ENTITY_REFS (${stopwatch.formatTime()}) ..."
+              )
 
-        // loop over entity types
-        def allTypesResult: Iterable[ReadWriteAction[Iterator[Int]]] =
-          entityTypeMetadata.map { case (entityType, metadata) =>
-            logger.info(s"Quicksilver migration:     - $entityType (${metadata.count}) ...")
+              /** *** don't delete legacy data; we'll do that en masse after everything is migrated
+              *  // delete legacy attributes from the ENTITY_ATTRIBUTE_xx_xx table
```

Comment:
Is this commented out code here for future reference or should it be deleted?

Reply:
I'm being very waffly here. When we discussed whether to clean out the legacy data as part of the migration, we didn't seem super confident. So I'm leaving this in place to make it super easy if we want to turn it back on. I could easily be convinced to just delete though.